### DISCUSSION TITLE
response code is 200 but response body is null

### DISCUSSION THREAD

**Discussion by Abhisheks1907** [fastapi/fastapi#6972] [Questions] [answered]
my main.py is
from fastapi import FastAPI, Request, Depends
from pydantic import BaseModel
import models
from sqlalchemy.orm import Session
from database import SessionLocal, engine
from models import product

app=FastAPI()

models.Base.metadata.create_all(bind=engine)

def get_db():
    try:
        db=SessionLocal()
        yield db
    finally:
        db.close()

@app.get('/')
def index():
    pass

@app.get('/products/{id}')
def get_product(id: int):
    db=SessionLocal()
    stock= db.query(models.product).filter(models.product.id == id).first()
    return stock

    # return pr.product_name ,pr.product_price,tenure,tenure*pr.product_price

and database.py is 
from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

SQLALCHEMY_DATABASE_URL = "sqlite:///./product.db"

engine = create_engine(
    SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)

Base = declarative_base()

and models.py is 
from sqlalchemy import  Column, ForeignKey, Integer, String

from database import Base

class product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    product_name = Column(String, unique=True, index=True)
    product_price= Column(Integer)

**Comment by krishnardt** [comment]
can you modify  this line with  in get_product function...
**db=SessionLocal()** --> **db = get_db()** and give it a try?
this would work.

**Comment by Abhisheks1907** [comment]
stock= db.query(models.product).filter(models.product.id == id).first()
this line it shows 
AttributeError: 'generator' object has no attribute 'query'

**Comment by Abhisheks1907** [comment]
solved

**Comment by krishnardt** [comment]
posting the solution about how you solved it would be better..

my suggestions:

```
@app.get('/products/{id}')
def get_product(id: int, db: Session = Depends(get_db)):
    stock= db.query(models.product).filter(models.product.id == id).first()
    return stock
```

or 

if you are using some function(not an api)  and that function is interacting with database...
```
def get_sample( db: Session = Depends(get_db)):
    """some crud operation"""
```

**Comment by TDoggin** [comment]
This maybe a long shot but can someone post the solution. I’m having the exact same issue with fast api.

**Comment by JarroVGIT** [comment, marked as answer]
This issue opening is a bunch of unformatted code without any question or issue in it. Somehow, someone did found what was wrong with it (kudos to him) and posted that solution in the comment above yours. This is a very specific user-error; it might be that the docs were not up to standards as they are now back then, but without any errors, there is little to go on here.

**Comment by TDoggin** [comment]
thanks for the feedback. i'm goign to open another issue then and see if i can get some help.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @Abhisheks1907  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
